### PR TITLE
Disallow `maxTimeMS` for non-awaitData cursors in `getMore` command

### LINKED ISSUE #2984: Support `maxTimeMS` in `getMore` command parameter
### What should be done?

After https://github.com/FerretDB/FerretDB/issues/2283

https://www.mongodb.com/docs/manual/reference/command/getMore/

`maxTimeMS` for `getMore` is supported only for tailable cursor.

### Where?

https://github.com/FerretDB/FerretDB/blob/main/internal/handlers/common/getmore.go

### Definition of Done

- all handlers updated;
- integration/compatibility tests added/updated;
- spot refactorings done;
- user documentation updated or an issue to create documentation created;

### PR DESCRIPTION
**Pull request by noisersup**
# Description

Closes #2984.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [x] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: integration/cursors/getmore_test.go [hunk 1]
```diff
@@ -952,45 +957,6 @@ func TestCursorsGetMoreCommandMaxTimeMSCursor(t *testing.T) {
 		integration.AssertMatchesCommandError(t, mongo.CommandError{Code: 50, Name: "MaxTimeMSExpired"}, cursor.Err())
 	})
 
-	t.Run("FindGetMoreMaxTimeMS", func(tt *testing.T) {
```
Comment (noisersup) [inline integration/cursors/getmore_test.go:955]: Those 2 test cases seem unnecessary. They mean to be resolved in #2983, which is about maxTimeMS propagation, however usage of maxTimeMS for non-awaitdata cursors is impossible, and it's resolved and tested in this PR

File: internal/handler/msg_getmore.go [hunk 2]
```diff
@@ -156,6 +160,14 @@ func (h *Handler) MsgGetMore(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
 		)
 	}
 
+	if maxTimeMSPresent && c.Type != cursor.TailableAwait {
```
Comment (rumyantseva) [inline internal/handler/msg_getmore.go:163]: Hm, strange that codecov shows the lines below as not covered

**Review by rumyantseva — APPROVED** [review]
LGTM

File: internal/handler/msg_getmore.go (hunk 2 above)
Comment (noisersup) [reply internal/handler/msg_getmore.go:163]: I had similar issue with previous PR, I think that codecov reports might be returned too early?

**Review by henvic — APPROVED** [review]

**Pull request merged**